src/Microsoft.TemplateEngine.Abstractions/ICacheParameter.cs

```diff
@@ -32,5 +32,10 @@ public interface ICacheParameter
         /// This property is localized if localizations are provided.
         /// </summary>
         string? Description { get; }
+
+        /// <summary>
+        /// Gets the default value to be used if the user specified the option without value.
+        /// </summary>
+        string? DefaultIfOptionWithoutValue { get; }
```

Comment:
How is this different than `DefaultValue` property above? Maybe we can make that a bit clear in the description.

Reply:
`Gets the default value to be used for parameter if the user did not specify the option.`
`Gets the default value to be used for parameter if the user specified the option without value.`
is it better like this?

Reply:
Thinking more I decided to drop `option` as this is CLI specific
Gets the default value to be used for parameter if the parameter is not passed for template instantiation.
Gets the default value to be used for parameter if the parameter is passed without value for template instantiation.